### PR TITLE
Increase test coverage for pkg/tracing to 100%

### PR DESCRIPTION
**Pull request by geeknoid**
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/411)

### CODE REVIEW

File: pkg/tracing/tracing_test.go [hunk 1]
```diff
@@ -113,6 +114,17 @@ func TestPropagateSpan(t *testing.T) {
 	}
 }
 
+func TestPropagateSpanError(t *testing.T) {
+	tracer := NewTracer(mocktracer.New())
+	tracer.inject = func(sm ot.SpanContext, format interface{}, carrier interface{}) error {
+		return errors.New("error")
```
Comment (ZackButcher) [inline pkg/tracing/tracing_test.go]: `t.Errorf`

Comment (geeknoid) [inline pkg/tracing/tracing_test.go]: This is a callback function that needs to return an error, so I think the code is right as written.

Comment (ZackButcher) [inline pkg/tracing/tracing_test.go]: Oh whoops, yup you're correct. Feels odd that nowhere in this test do we have a failure condition - how do we know short of a panic that things are working as intended?

File: pkg/tracing/tracing.go [hunk 2]
```diff
@@ -67,16 +67,27 @@ type Tracer struct {
 	ot.Tracer
 
 	enabled bool
+
+	// this indirection exists strictly to enable error injection
+	inject func(sm ot.SpanContext, format interface{}, carrier interface{}) error
```
Comment (ZackButcher) [inline pkg/tracing/tracing.go]: I don't love adding stuff like this just to we can tickle code paths for test coverage.

File: pkg/tracing/tracing_test.go (hunk 1 above)
Comment (geeknoid) [inline pkg/tracing/tracing_test.go]: The comment at the end of the function says it all...

This is indeed a simple test to ensure that the logging statement in the failure handling code doesn't lead to a crash (which unexercised failure handling path are prone to do)

File: pkg/tracing/tracing.go (hunk 2 above)
Comment (geeknoid) [inline pkg/tracing/tracing.go]: I want us to be at 100% test coverage in the near future. I'm happy to hear of alternative approaches to do that in this case.

I consider this  a pretty cheap and expedient way to get the job done without having to create a whole bunch of mock infrastructure.

**Pull request merged**